org.civicrm.reqangular/js/test/models/instances/contact-instance.spec.js

```diff
@@ -3,35 +3,81 @@
 define([
   'common/lodash',
   'common/angularMocks',
+  'common/mocks/services/api/contact-mock',
   'common/models/instances/contact-instance'
 ], function (_) {
   'use strict';
 
   describe('ContactInstance', function () {
-    var ContactInstance, ModelInstance, ContactAPI;
+    var $provide, $rootScope, ContactInstance, ModelInstance, ContactAPI;
 
-    beforeEach(module('common.models.instances'));
-    beforeEach(inject(['api.contact', 'ContactInstance', 'ModelInstance',
-      function (_ContactAPI_, _ContactInstance_, _ModelInstance_) {
+    beforeEach(function () {
+      module('common.models.instances', 'common.mocks', function (_$provide_) {
+        $provide = _$provide_;
+      });
+      inject(['api.contact.mock', function (contactAPIMock) {
+        $provide.value('api.contact', contactAPIMock);
+      }]);
+    });
+
+    beforeEach(inject(['$rootScope', 'api.contact', 'ContactInstance', 'ModelInstance',
+      function (_$rootScope_, _ContactAPI_, _ContactInstance_, _ModelInstance_) {
+        $rootScope = _$rootScope_;
         ContactInstance = _ContactInstance_;
         ModelInstance = _ModelInstance_;
         ContactAPI = _ContactAPI_;
       }
     ]));
 
-    beforeEach(function () {
-      spyOn(ContactAPI, 'leaveManagees');
-    });
-
     it('inherits from ModelInstance', function () {
       expect(_.functions(ContactInstance)).toEqual(jasmine.arrayContaining(_.functions(ModelInstance)));
     });
 
-    describe('leaveManagees()', function() {
+    describe('checkIfSelfLeaveApprover()', function () {
+      var callCheckIfSelfLeaveApprover, result;
+
+      beforeEach(function () {
+        callCheckIfSelfLeaveApprover = function () {
+          ContactInstance.checkIfSelfLeaveApprover()
+            .then(function (_result_) {
+              result = _result_;
+            });
+          $rootScope.$digest();
+        };
+      });
+
+      describe('when leave managees of the contact contain the contact', function () {
+        beforeEach(function () {
+          ContactInstance.id = '1';
```

Comment:
This is because contact `1` has managees while `1010011010` does not have (it is a random ID). Would you like me to mock the getmanagees method instead?

Reply:
Yes please. Currently if the contacts data changes, there is a possibility of this test being broken. So instead if we mock `leaveManagees` function and tell it return either of loggedInId or some thing else, it would be ideal.

Reply:
Done.